src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5058,6 +5064,10 @@ int wm_vuldet_collect_agent_software(scan_agent *agent, sqlite3 *db, scan_ctx_t*
                     wm_vuldet_free_cpe(&s_cpe);
                 }
 
+                if (win_architecture) {
```

Comment:
The `os_free` macro includes a `NULL` check so, this `if` statement is not necessary.

Reply:
Fixed!